lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp

```diff
@@ -1910,26 +1908,72 @@ void mlir::torch::onnx_c::populateDefaultDomainQtoZ(
           }
         }
 
-        // Handle cases that axes are explicitly specified.
-        // Extract the axes values from the axes operand.
-        // This really shouldn't happen but it helps pass weird tests.
-        // TODO: Derive the chosen axes from the data type and final result type
-        // instead of using the dynamic axes at operand[1].
         if (!binder.tensorOperandAtIndex(axes, 1)) {
-          Torch::BaseTensorType axesType =
-              axes.getType().cast<Torch::BaseTensorType>();
-          auto sizes = axesType.getSizes();
-          for (int i = 0; i < sizes[0]; i++) {
-            Value selectIndex = rewriter.create<Torch::ConstantIntOp>(
-                binder.getLoc(), rewriter.getI64IntegerAttr(i));
-            Value extract = rewriter.create<Torch::AtenSelectIntOp>(
-                binder.getLoc(),
-                axesType.getWithSizesAndDtype(llvm::SmallVector<int64_t>{1},
-                                              axesType.getOptionalDtype()),
-                axes, zero, selectIndex);
-            Value dim = rewriter.create<Torch::AtenItemOp>(binder.getLoc(),
-                                                           torchIntTy, extract);
-            axesList.push_back(dim);
+          SmallVector<int64_t> inputShape{dataTy.getSizes()};
+          SmallVector<int64_t> resultShape{resultType.getSizes()};
+          // if the shapes are equal, none of the dims is reduced
+          if (llvm::equal(inputShape, resultShape)) {
```

Comment:
We cannot guarantee this is true. Dynamic shapes with poor shape inference could result in multiple dynamic dimensions with currently unknown folding.